src/main/java/com/nedap/archie/flattener/Flattener.java

```diff
@@ -498,6 +498,35 @@ private void flattenCComplexObject(CComplexObject newObject, CComplexObject spec
         for(CAttribute attribute:specialized.getAttributes()) {
             flattenSingleAttribute(newObject, attribute);
         }
+        for(CAttributeTuple tuple:specialized.getAttributeTuples()) {
+            flattenTuple(newObject, tuple);
+        }
+    }
+
+    private void flattenTuple(CComplexObject newObject, CAttributeTuple tuple) {
+        CAttributeTuple matchingTuple = findMatchingTuple(newObject.getAttributeTuples(), tuple);
+
+
+        CAttributeTuple tupleClone = (CAttributeTuple) tuple.clone();
+        if(matchingTuple == null) {
+            //add
+            newObject.addAttributeTuple(tupleClone);
+        } else {
+            //replace
+            newObject.getAttributeTuples().remove(matchingTuple);
+            newObject.addAttributeTuple(tupleClone);
```

Comment:
does the order matter in which the tuples are added? ie. does the tupleClone need to be added at the same place as where the matchingTuple was?

Reply:
I think the ordering of tuples should not matter much - they are alternatives, not different elements to be added. Perhaps in a generated UI it does matter. I don't think we need to change this.

Reply:
Alright, we can tackle this if we run into this